collector/collector/lib/validation.py

```diff
@@ -0,0 +1,173 @@
+    'Arch',
+    'Build',
+    'Creation-Timestamp',
+    'Classification',
+    'Host-Type',
+    'Kernel-Version',
+    'Machine-Id',
+    'Severity',
+    'Record-Format-Version',
+)
+
+REQUIRED_HEADERS_V2 = REQUIRED_HEADERS_V1 + (
+    'Payload-Format-Version',
+    'System-Name',
+    'X-Telemetry-Tid',
+)
+
+REQUIRED_HEADERS_V3 = REQUIRED_HEADERS_V2 + (
+    'Board-Name',
+    'Bios-Version',
+    'Cpu-Model',
+)
+
+# see config.py for the meaning of this value
+TELEMETRY_ID = app.config.get("TELEMETRY_ID", "6907c830-eed9-4ce9-81ae-76daf8d88f0f")
+
+SEVERITY_VALUES = [1, 2, 3, 4]
+VALID_RECORD_FORMAT_VERSIONS = [1, 2, 3]
+POSTGRES_INT_MAX = 2147483647
+MAXLEN_PRINTABLE = 200
+
+
+class InvalidUsage(Exception):
+    status_code = 400
+
+    def __init__(self, message, status_code=None, payload=None):
+        Exception.__init__(self)
+        self.message = message
+        if status_code is not None:
+            self.status_code = status_code
+        self.payload = payload
+        app.logger.error("InvalidUsage ({}): {}".format(self.status_code, self.message))
+
+    def to_dict(self):
+        rv = dict(self.payload or ())
+        rv['message'] = self.message
+        return rv
+
+    def __str__(self):
+        return self.message
+
+
+def validate_headers(headers, required_headers):
+    """ Check for every single required header to be
+        in the request """
+    req_headers = dict(headers)
+    req_headers_keys = req_headers.keys()
+    for header in required_headers:
+        if header not in req_headers_keys:
+            return False
+    return True
+
+
+def record_format_version_validation(record_format_version):
+    return record_format_version and int(record_format_version) in VALID_RECORD_FORMAT_VERSIONS
+
+
+def record_format_version_headers_validation(record_format_version, headers):
+    # Validate required headers based on Record Version
+    if int(record_format_version) == 1:
+        return validate_headers(headers, REQUIRED_HEADERS_V1)
+    elif int(record_format_version) == 2:
+        return validate_headers(headers, REQUIRED_HEADERS_V2)
+    elif int(record_format_version) == 3:
+        return validate_headers(headers, REQUIRED_HEADERS_V3)
+    return False
+
+
+def value_is_printable(a_value):
+    return all([x in string.printable for x in a_value])
+
+
+def validation_tid_header(tid):
+    return tid == TELEMETRY_ID
+
+
+def validate_severity(severity):
+    return severity not in SEVERITY_VALUES
+
+
+def validate_classification(classification):
+    return classification is not None and len(classification.split('/')) == 3
+
+
+def validate_machine_id(machine_id):
+    return machine_id is not None and len(machine_id) <= 32
+
+
+def validate_timestamp(timestamp):
+    """ Verify that the ts is at least > 01/01/2017"""
+    return timestamp is not None and \
+           str(timestamp).isdigit() and \
+           int(timestamp) > 1483232400 & len(str(timestamp)) == 10 & str(timestamp).isdigit()
```

Comment:
Any reason for restricting the timestamp like this? Should we just check to make sure it is an integer that will satisfy the type of the field in the database?